app/models/claim/matching_attribute_finder.rb

```diff
@@ -15,17 +13,29 @@ def initialize(source_claim, claims_to_compare = Claim.submitted)
     end
 
     def matching_claims
-      predicates = []
-      values = []
-      ATTRIBUTES_TO_MATCH.each do |attribute|
-        value = @source_claim.read_attribute(attribute)
-        next if value.blank?
-
-        predicates << "LOWER(#{attribute}) = LOWER(?)"
-        values << value
+      claims = @claims_to_compare.where.not(id: @source_claim.id)
+      match_queries = nil
+
+      ATTRIBUTE_GROUPS_TO_MATCH.each do |attributes|
+        vals = values_for_attributes(attributes)
+
+        next if vals.blank?
```

Comment:
`empty?` would be more appropriate for an array as it will signal to the reader what the type of `vals` is. `blank?` is more for values that are expected to be a string.